### ISSUE TITLE
Add node18 support for @apollo/federation

### ISSUE THREAD

**Issue by progfay** [apollographql/federation#2167]
- The Active LTS support of Node.js v16 will be end in 2022-10-17
  - ref. https://github.com/nodejs/release#release-schedule
- However, the latest version of `@apollo/federation` () does not support Node.js v18 yet...
  - related: https://github.com/apollographql/federation/issues/1752
- This causes an error when trying to install a package that depends on @apollo/federation in Node.js v18 (e.g.: [`apollo`](https://www.npmjs.com/package/apollo))

```console
❯ node -v
v18.9.1

❯ yarn add @apollo/federation@latest
yarn add v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error @apollo/subgraph@0.5.1: The engine "node" is incompatible with this module. Expected version ">=12.13.0 <18.0". Got "18.9.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
zsh: exit 1     yarn add @apollo/federation@latest

❯ yarn add apollo@latest
yarn add v1.22.19
info No lockfile found.
[1/4] 🔍  Resolving packages...
warning apollo > cli-ux@6.0.9: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
[2/4] 🚚  Fetching packages...
error @apollo/federation@0.27.0: The engine "node" is incompatible with this module. Expected version ">=12.13.0 <17.0". Got "18.9.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
zsh: exit 1     yarn add apollo@latest
```

To resolve this problem, We need to add Node.js v18 support for `@apollo/federation`.

**Comment by fbartho** [comment]
I too am tripping on this now! Thanks @jeffjakub

**Comment by rmarku** [comment]
I hope this is taken care soon

**Comment by glasser** [comment]
This has largely been fixed in `@apollo/federation@0.38.1`, which updates its dependency on `@apollo/subgraph` to `^0.26.1`, and that version of `@apollo/subgraph` drops its statement of incompatibility with Node v18.

I see that you are concerned as well about the fact that the `apollo` CLI depends strictly on an even older version of `@apollo/federation`. That said, the `apollo` CLI (which has been mostly deprecated for a while) has never as a whole been tested with Node 18... so either you can decide for yourself that it probably works with Node 18 (and use `yarn --ignore-engines` or whatever), or you have deeper issues using `apollo` with Node 18 than the fact that one particular package happens to have a `<18` written in its package.json (which we only put there due to an old habit of always putting `<` constraints on our packages, not because we actually knew that it was incompatible with Node 18).

We are planning to fully end-of-life the `apollo` CLI by the end of the year by providing clearly documented non-bitrotten alternatives for each of its features; until then, I encourage you to stick to Node 14 or 16 when using this CLI.  (While you're right that Node 16 is not in *Active* LTS any more, it remains in *Maintenance* LTS until next September.)

**Comment by StanislavPavlov** [comment]
@glasser 
Is this the only solution for today? It doesn't seem to be working for me.
We recently upgraded expo, and as a consequence we had to upgrade other dependencies that require Node 18. This is now a problem for us, as we cannot release a new build that requires App Store Connect.
